### PR TITLE
Al/upgrade eks module from 17.24.0 to 19.10.0

### PR DESCRIPTION
**Pull request by alouvion-aneo**
goal is to upgrade eks module from 17 to 19
module webpage : https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest

### CODE REVIEW

File: infrastructure/modules/aws/ecr/versions.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.18.0"
+      version = ">= 4.47"
```
Comment (lemaitre-aneo) [inline infrastructure/modules/aws/ecr/versions.tf:5]: Is this the minimal version where force_delete is supported? or just the latest version?

File: infrastructure/modules/aws/eks/iam.tf [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
-resource "aws_iam_role_policy_attachment" "ssm_agent" {
-  policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
-  role       = module.eks.worker_iam_role_name
-}
\ No newline at end of file
+# resource "aws_iam_role_policy_attachment" "ssm_agent" {
+#   for_each = module.eks.self_managed_node_groups
+#   policy_arn = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+#   role       = each.value.iam_role_name
+# }
```
Comment (lemaitre-aneo) [inline infrastructure/modules/aws/eks/iam.tf:6]: I'm not sure we want to keep it commented. I would suggest to remove it entirely

File: infrastructure/modules/aws/eks/instance_refresh.tf [hunk 3]
```diff
@@ -137,12 +148,16 @@ resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_spot" {
...
+  #count                  = length(module.eks.self_managed_node_groups_autoscaling_group_names)
+  count                  = length(local.eks_worker_group)
   name                   = "aws-node-termination-handler"
-  autoscaling_group_name = module.eks.workers_asg_names[count.index]
+  autoscaling_group_name = module.eks.self_managed_node_groups_autoscaling_group_names[count.index]
   lifecycle_transition   = "autoscaling:EC2_INSTANCE_TERMINATING"
   heartbeat_timeout      = 300
   default_result         = "CONTINUE"
+  depends_on = [
+  module.eks
```
Comment (lemaitre-aneo) [inline infrastructure/modules/aws/eks/instance_refresh.tf:159]: Do you really need this?

File: infrastructure/modules/aws/eks/patch-coredns.tf [hunk 4]
```diff
@@ -2,9 +2,6 @@
 resource "null_resource" "patch_coredns" {
   provisioner "local-exec" {
     command = "kubectl -n kube-system patch deployment coredns --patch \"${yamlencode(local.patch_coredns_spec)}\""
-    environment = {
-      KUBECONFIG = module.eks.kubeconfig_filename
```
Comment (lemaitre-aneo) [inline infrastructure/modules/aws/eks/patch-coredns.tf:6]: Why remove this? I think deployment must work even if we do not update the `~/.kube/config` file

File: infrastructure/modules/aws/eks/versions.tf [hunk 5]
```diff
@@ -2,19 +2,19 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 4.18.0"
+      version = ">= 4.47.0"
     }
     kubernetes = {
       source  = "hashicorp/kubernetes"
-      version = ">= 2.7.1"
+      version = "~> 2.13.0"
```
Comment (lemaitre-aneo) [inline infrastructure/modules/aws/eks/versions.tf:9]: Prefer `>=`: https://developer.hashicorp.com/terraform/language/expressions/version-constraints#terraform-core-and-provider-versions

File: infrastructure/quick-deploy/aws/all/parameters.tfvars [hunk 6]
```diff
@@ -41,35 +41,39 @@ eks = {
...
-    asg_min_size                             = 1
-    asg_max_size                             = 5
-    asg_desired_capacity                     = 1
-    on_demand_base_capacity                  = 1
-    on_demand_percentage_above_base_capacity = 100
-    kubelet_extra_args                       = "--node-labels=grid/type=Operator --register-with-taints=grid/type=Operator:NoSchedule"
-  }
-]
+eks_operational_worker_groups = {
+    eks_operational_worker_groups = {
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/parameters.tfvars:45]: Why the key is the same as the variable name?

File: infrastructure/quick-deploy/aws/all/providers.tf [hunk 7]
```diff
@@ -5,18 +5,30 @@ provider "aws" {
...
-  token                  = module.eks.token
-  insecure               = false
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", module.eks.cluster_name]
+  }
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/providers.tf:15]: Why did you need to change?

File: infrastructure/modules/aws/ecr/versions.tf (hunk 1 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/ecr/versions.tf:5]: I took 4.47 version because of aws eks module v19 : https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v19.10.0/versions.tf#L7

File: infrastructure/modules/aws/eks/iam.tf (hunk 2 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/iam.tf:6]: yes sure it was only intended for helping the review

File: infrastructure/modules/aws/eks/instance_refresh.tf (hunk 3 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/instance_refresh.tf:159]: no

File: infrastructure/modules/aws/eks/patch-coredns.tf (hunk 4 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/patch-coredns.tf:6]: It has been removed in v18 https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/UPGRADE-18.0.md look for : kubeconfig_filename

File: infrastructure/modules/aws/eks/update-kubeconfig.tf [hunk 8]
```diff
@@ -4,7 +4,7 @@ resource "null_resource" "update_kubeconfig" {
     cluster_arn = module.eks.cluster_arn
   }
   provisioner "local-exec" {
-    command = "aws --profile ${var.profile} eks update-kubeconfig --region ${local.region} --name ${var.name}"
+    command = "aws --profile ${var.profile} eks update-kubeconfig --region ${local.region} --name ${module.eks.cluster_name}"
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/update-kubeconfig.tf:7]: good point !

File: infrastructure/modules/aws/eks/variables.tf [hunk 9]
```diff
@@ -7,7 +7,7 @@ variable "profile" {
 # Tags
 variable "tags" {
   description = "Tags for resource"
-  type        = any
+  type        = map(string)
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/variables.tf:10]: target of tag is for eks module so I prefered restrict the type for the target usage as seen in aws eks module v19 : 
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v19.10.0/variables.tf#L9

File: infrastructure/modules/aws/eks/versions.tf (hunk 5 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/versions.tf:9]: I agree

File: infrastructure/quick-deploy/aws/all/parameters.tfvars (hunk 6 above)
Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/all/parameters.tfvars:45]: node_group was an array now it is a map
for the moment we have cut in 2 maps :
operational worker => eks_operational_worker_groups
workers (armonik) =>eks_worker_groups
As spoken with Lilia next step in the refacto will be to join the 2 maps and we will nomore need the 2 maps.

File: infrastructure/quick-deploy/aws/all/providers.tf (hunk 7 above)
Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/all/providers.tf:15]: I followed the official included exemple from the module
in v17 it was : https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v17.24.0/examples/managed_node_groups/main.tf#L102
in v19 it is now : https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v19.10.0/examples/self_managed_node_group/main.tf#L5

File: infrastructure/quick-deploy/aws/all/versions.tf [hunk 10]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 4.18.0"
+      version = ">= 4.47.0"
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/all/versions.tf]: ok !

File: infrastructure/modules/aws/eks/iam.tf (hunk 2 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/iam.tf]: removed

File: infrastructure/quick-deploy/aws/all/parameters.tfvars (hunk 6 above)
Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/all/parameters.tfvars]: renamed

File: infrastructure/quick-deploy/aws/all/armonik.tf [hunk 11]
```diff
@@ -57,15 +57,4 @@ module "armonik" {
...
-    kubernetes_secret.fluent_bit,
-    kubernetes_secret.grafana,
-    kubernetes_secret.metrics_exporter,
-    kubernetes_secret.partition_metrics_exporter,
-    kubernetes_secret.seq,
-    kubernetes_secret.shared_storage,
-    kubernetes_secret.deployed_object_storage,
-    kubernetes_secret.deployed_table_storage,
-    kubernetes_secret.deployed_queue_storage,
-  ]
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/all/armonik.tf:70]: Why is this part of this PR?

File: infrastructure/quick-deploy/aws/eks/providers.tf [hunk 12]
```diff
@@ -5,24 +5,46 @@ provider "aws" {
...
+#   helm_driver = "configmap"
+#   kubernetes {
+#     host                   = module.eks.cluster_endpoint
+#     cluster_ca_certificate = base64decode(module.eks.certificate_authority.0.data)
+#     token                  = module.eks.token
+#     #config_path            = pathexpand("~/.kube/config")
+#     insecure = false
+#   }
+# }
+
```
Comment (lemaitre-aneo) [inline infrastructure/quick-deploy/aws/eks/providers.tf:47]: Either delete or uncomment

File: infrastructure/quick-deploy/aws/all/armonik.tf (hunk 11 above)
Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/all/armonik.tf]: it has been wrongly included during my fix format commit

File: infrastructure/quick-deploy/aws/eks/providers.tf (hunk 12 above)
Comment (alouvion-aneo) [reply infrastructure/quick-deploy/aws/eks/providers.tf]: done

File: infrastructure/modules/aws/eks/eks-cluster.tf [hunk 13]
```diff
@@ -1,78 +1,86 @@
...
+  cloudwatch_log_group_kms_key_id        = var.eks.encryption_keys.cluster_log_kms_key_id
+  cloudwatch_log_group_retention_in_days = var.eks.cluster_log_retention_in_days
+  create_cluster_security_group          = true
+  cluster_encryption_config = {
+    provider_key_arn = var.eks.encryption_keys.cluster_encryption_config
+    resources        = ["secrets"]
+  }
 
   # Tags
+  #TODO: Arnaud.L tags must be added back - pb with depends_on on module 'bug'
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/eks-cluster.tf:38]: Delete the comment

File: infrastructure/modules/aws/eks/eks-cluster.tf [hunk 14]
```diff
@@ -1,78 +1,86 @@
...
   tags         = local.tags
   cluster_tags = local.tags
 
   # IAM
-  map_roles = concat([
+  # used to allow other users to interact with our cluster
+  aws_auth_roles = concat([
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/eks-cluster.tf:44]:
```suggestion
  aws_auth_roles = var.eks.map_roles
```

File: infrastructure/modules/aws/eks/eks-cluster.tf [hunk 15]
```diff
@@ -1,78 +1,86 @@
...
     {
-      rolearn  = module.eks.worker_iam_role_arn
+      rolearn  = local.cluster_iam_role_arn
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/eks-cluster.tf:46]: It is the iam_role_arn of each self_managed_node_group

File: infrastructure/modules/aws/eks/eks-cluster.tf [hunk 16]
```diff
@@ -1,78 +1,86 @@
...
+      "k8s.io/cluster-autoscaler/enabled" : true,
+      "k8s.io/cluster-autoscaler/${var.name}" : "owned"
+      "aws-node-termination-handler/managed" : true
+    }
+  }
+
+  # it replaces previously created role in iam.tf 
+  iam_role_additional_policies = {
+    AmazonSSMManagedInstanceCore = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+    # "eks-vpc-resource-access" = "arn:aws:iam::aws:policy/AmazonEKSVPCResourceController"
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/eks-cluster.tf:71]: uncomment  or delete the comment

File: infrastructure/modules/aws/eks/eks-cluster.tf [hunk 17]
```diff
@@ -1,78 +1,86 @@
...
+  # Worker groups
+  self_managed_node_groups = local.eks_worker_group
   /*
+  module input from doc : https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/latest?tab=inputs#optional-inputs
+  variables from module code : https://github.dev/terraform-aws-modules/terraform-aws-eks/tree/v19.10.0
+  sample usages : https://github.com/Jitsusama/example-terraform-eks-mixed-os-cluster/blob/main/cluster.tf#L91
+                  https://github.dev/terraform-aws-modules/terraform-aws-eks/tree/v19.10.0
+
+
+
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/eks-cluster.tf:83]: Put these comments before : self_managed_node_groups = local.eks_worker_group

File: infrastructure/modules/aws/eks/locals.tf [hunk 18]
```diff
@@ -74,47 +74,54 @@ locals {
...
+          EOT
+          metadata_http_endpoint               = "enabled"  # The state of the metadata service: enabled, disabled.
+          metadata_http_tokens                 = "required" # If session tokens are required: optional, required.
+          metadata_http_put_response_hop_limit = 2
+          # The desired HTTP PUT response hop limit for instance metadata requests.
+          # tags = merge(try(v.tags, {}), {
+          #   "k8s.io/cluster-autoscaler/enabled" = true
+          #   "k8s.io/cluster-autoscaler/${var.name}" = true
+          #   "aws-node-termination-handler/managed" = true
+          # })
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/locals.tf:99]: Delete comments

File: infrastructure/modules/aws/eks/locals.tf [hunk 19]
```diff
@@ -74,47 +74,54 @@ locals {
...
+        echo fs.inotify.max_user_instances=8192 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+        EOT
+          metadata_http_endpoint               = "enabled"  # The state of the metadata service: enabled, disabled.
+          metadata_http_tokens                 = "required" # If session tokens are required: optional, required.
+          metadata_http_put_response_hop_limit = 2
+          # tags = merge(try(v.tags, {}), {
+          #   "k8s.io/cluster-autoscaler/enabled" = true
+          #   "k8s.io/cluster-autoscaler/${var.name}" = "owned"
+          #   "aws-node-termination-handler/managed" = true
+          # })
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/locals.tf:123]: delete comments

File: infrastructure/modules/aws/eks/instance_refresh.tf [hunk 20]
```diff
@@ -137,9 +148,9 @@ resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_spot" {
 # ensures that node termination does not require the lifecycle action to be completed,
 # and thus allows the ASG to be destroyed cleanly.
 resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler" {
-  count                  = length(module.eks.workers_asg_names)
+  count                  = length(local.eks_worker_group)
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/instance_refresh.tf:151]:
```suggestion
  for_each                  = toset(module.eks.self_managed_node_groups_autoscaling_group_names)
```

File: infrastructure/modules/aws/eks/outputs.tf [hunk 21]
```diff
@@ -32,19 +37,28 @@ output "kms_key_id" {
...
+
+# TODO: ARNAUD.L : we must return list of iam_role_name,
+# 1 for each group in : self_managed_node_groups
+# and maybe todo later : 1 more for each group in eks_managed_node_groups
 output "worker_iam_role_name" {
   description = "EKS worker IAM role name"
-  value       = module.eks.worker_iam_role_name
+  #value       = module.eks.self_managed_node_groups["worker-c5.4xlarge-spot"].iam_role_name
+  #value =       try(values(module.eks.self_managed_node_groups).1.iam_role_name, "")
+  value = module.eks.cluster_iam_role_name
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/outputs.tf:48]: Create two outputs:
* worker_iam_role_name: list of iam_role_name of all sef_managed_node_group
* cluster_iam_role_name or eks_iam_role_name: value = module.eks.cluster_iam_role_name

File: infrastructure/modules/aws/eks/outputs.tf [hunk 22]
```diff
@@ -32,19 +37,28 @@ output "kms_key_id" {
...
+output "self_managed_node_groups" {
+  value = module.eks.self_managed_node_groups
 }
 
-output "kubectl_config_filename" {
-  description = "kubectl config as generated by the module"
-  value       = module.eks.kubeconfig_filename
+output "cluster_id" {
+  description = "EKS cluster ID"
+  value       = module.eks.cluster_name
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/outputs.tf:61]:
```suggestion
  value       = module.eks.cluster_id
```

File: infrastructure/modules/aws/eks/outputs.tf [hunk 23]
```diff
@@ -1,26 +1,31 @@
...
-output "token" {
-  description = "Authentication token for EKS"
-  value       = data.aws_eks_cluster_auth.cluster.token
-  sensitive   = true
+output "cluster_name" {
+  description = "cluster_name"
+  value       = module.eks.cluster_name
 }
 
 output "name" {
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/outputs.tf:26]:
```suggestion
output "cluster_name" {
```

Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/outputs.tf:26]: Delete this output (cf. cluster_name)

File: infrastructure/quick-deploy/aws/all/parameters.tfvars [hunk 24]
```diff
@@ -41,35 +41,39 @@ eks = {
...
 
 # EKS worker groups
-eks_worker_groups = [
-  {
+eks_worker_groups = {
+  linux = {
     name                                     = "worker-c5.4xlarge-spot"
     spot_allocation_strategy                 = "capacity-optimized"
-    override_instance_types                  = ["c5.4xlarge"]
+    instance_type                            = "c5.4xlarge"
```
Comment (lzianekhodja) [inline infrastructure/quick-deploy/aws/all/parameters.tfvars:65]: Use a list of instance types (see the example in Terraform EKS module version 1.19.0)

File: infrastructure/quick-deploy/aws/eks/providers.tf [hunk 25]
```diff
@@ -5,47 +5,30 @@ provider "aws" {
...
+# package manager for kubernetes
 provider "helm" {
   helm_driver = "configmap"
   kubernetes {
-    token    = module.eks.token
-    host     = module.eks.cluster_endpoint
-    insecure = true
+    host                   = module.eks.cluster_endpoint
+    cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+    #token                  = module.eks.token
```
Comment (lzianekhodja) [inline infrastructure/quick-deploy/aws/eks/providers.tf:24]: Delete comment

File: infrastructure/modules/aws/eks/eks-cluster.tf (hunk 14 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/eks-cluster.tf:44]: ok removed because already done by aws eks moduls itself

File: infrastructure/modules/aws/eks/eks-cluster.tf (hunk 15 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/eks-cluster.tf:46]: ok removed because already done by aws eks moduls itself

File: infrastructure/modules/aws/eks/instance_refresh.tf (hunk 20 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/instance_refresh.tf:151]: ok nice

File: infrastructure/modules/aws/eks/instance_refresh.tf [hunk 26]
```diff
@@ -137,9 +148,9 @@ resource "aws_cloudwatch_event_rule" "aws_node_termination_handler_spot" {
...
   name                   = "aws-node-termination-handler"
-  autoscaling_group_name = module.eks.workers_asg_names[count.index]
+  autoscaling_group_name = module.eks.self_managed_node_groups_autoscaling_group_names[count.index]
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/instance_refresh.tf:153]: same modif than previous comment. nice

File: infrastructure/modules/aws/eks/outputs.tf (hunk 21 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/outputs.tf:48]: ok

File: infrastructure/modules/aws/eks/outputs.tf (hunk 22 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/outputs.tf:61]: ok so now we have both : name and id

File: infrastructure/modules/aws/eks/outputs.tf (hunk 23 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/outputs.tf:26]: ok

Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/outputs.tf:26]: hum not sure do we need both cluster_name and cluster_id ? or only one of them ?

File: infrastructure/modules/aws/eks/instance_refresh.tf (hunk 20 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/instance_refresh.tf:151]: hum can't : 
│ Error: Invalid for_each argument
│
│   on ../../../modules/aws/eks/instance_refresh.tf line 151, in resource "aws_autoscaling_lifecycle_hook" "aws_node_termination_handler":
│  151:   for_each               = toset(module.eks.self_managed_node_groups_autoscaling_group_names)
│     ├────────────────
│     │ module.eks.self_managed_node_groups_autoscaling_group_names is a list of string, known only after apply

File: infrastructure/modules/aws/eks/eks-cluster.tf [hunk 27]
```diff
@@ -1,78 +1,86 @@
+locals {
+  cluster_iam_role_arn = module.eks.cluster_iam_role_arn
+}
```
[the remark this answers is not in the document]
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/eks-cluster.tf]: removed

File: infrastructure/modules/aws/eks/outputs.tf [hunk 28]
```diff
@@ -1,28 +1,34 @@
...
-  description = "Name of EKS cluster"
-  value       = module.eks.cluster_id
+output "cluster_name" {
+  description = "EKS cluster name"
+  value       = module.eks.cluster_name
 }
 
+output "cluster_id" {
+  description = "EKS cluster ID  used for backword compatibility : https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/UPGRADE-19.0.md#list-of-backwards-incompatible-changes"
+  value       = module.eks.cluster_name
```
Comment (lzianekhodja) [inline infrastructure/modules/aws/eks/outputs.tf:28]:
```suggestion
  value       = module.eks.cluster_id
```

File: infrastructure/modules/aws/eks/outputs.tf (hunk 21 above)
Comment (lzianekhodja) [reply infrastructure/modules/aws/eks/outputs.tf:46]: Not yet corrected

File: infrastructure/quick-deploy/aws/eks/outputs.tf [hunk 29]
```diff
@@ -2,10 +2,10 @@
...
   description = "EKS parameters"
   value = {
-    name                    = module.eks.name
-    cluster_id              = module.eks.cluster_id
-    worker_iam_role_name    = module.eks.worker_iam_role_name
-    kubectl_config_filename = abspath(module.eks.kubectl_config_filename)
-    issuer                  = module.eks.issuer
+    name                               = module.eks.cluster_name
+    cluster_id                         = module.eks.cluster_id
+    worker_iam_role_name               = module.eks.worker_iam_role_name
```
Comment (lzianekhodja) [inline infrastructure/quick-deploy/aws/eks/outputs.tf:7]: as defined in the output of EKS module, it is the cluster role name.

**Review by lzianekhodja — COMMENTED** [review]
You must replace worker_iam_role_name in infrastructure/quick-deploy/aws/monitoring/iam.tf

File: infrastructure/modules/aws/eks/outputs.tf (hunk 21 above)
Comment (alouvion-aneo) [reply infrastructure/modules/aws/eks/outputs.tf]: worker_iam_role_name deleted

**Review by lzianekhodja — APPROVED** [review]
